code/modules/mob/living/basic/space_fauna/eyeball/eyeball_ai_subtree.dm

```diff
@@ -0,0 +1,54 @@
+/datum/ai_controller/basic_controller/eyeball
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/eyeball,
+		BB_EYE_DAMAGE_THRESHOLD = 10,
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/targeted_mob_ability/glare,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/heal_the_blind,
+		/datum/ai_planning_subtree/find_and_hunt_target/carrot,
+	)
+
+/datum/ai_planning_subtree/heal_the_blind
+
+/datum/ai_planning_subtree/heal_the_blind/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	var/mob/living/carbon/target = controller.blackboard[BB_BLIND_TARGET]
+
+	if(QDELETED(target))
+		controller.queue_behavior(/datum/ai_behavior/find_the_blind, BB_BLIND_TARGET, BB_EYE_DAMAGE_THRESHOLD)
+		return
+
+	controller.queue_behavior(/datum/ai_behavior/heal_eye_damage, BB_BLIND_TARGET)
+	return SUBTREE_RETURN_FINISH_PLANNING
+
+/datum/targetting_datum/basic/eyeball/can_attack(mob/living/owner, atom/target)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!ishuman(target))
+		return TRUE
+	var/mob/living/carbon/human_target = target
+	var/eye_damage_threshold = owner.ai_controller.blackboard[BB_EYE_DAMAGE_THRESHOLD]
+	if(!eye_damage_threshold)
+		return TRUE
+	var/obj/item/organ/internal/eyes/eyes = human_target.get_organ_slot(ORGAN_SLOT_EYES)
+	if(eyes.damage > eye_damage_threshold) //we dont attack people with bad vision
+		return FALSE
```

Comment:
Should this also check `human_target.is_blind()` or does `can_see` do that?

Reply:
yes